File: mobility_reports/mobility_reports/report/splitted_report/splitted_report.py
# Copyright (c) 2025, Aerele Technologies Private Limited and contributors
# For license information, please see license.txt

import frappe

def execute(filters=None):
    data = {
        "sales_orders": get_sales_orders(filters),
        "sales_invoices": get_sales_invoices(filters)
    }
    
    return data

def get_sales_orders(filters):
    return frappe.db.sql("""
        SELECT 
            name, customer, total 
        FROM `tabSales Order`
        WHERE docstatus = 1
    """, as_dict=True)

def get_sales_invoices(filters):
    return frappe.db.sql("""
        SELECT 
            name, customer, grand_total 
        FROM `tabSales Invoice`
        WHERE docstatus = 1
    """, as_dict=True)

File: mobility_reports/mobility_reports/report/splitted_report/splitted_report.js
// Copyright (c) 2025, Aerele Technologies Private Limited and contributors
// For license information, please see license.txt


frappe.query_reports["Sales & Invoices Report"] = {
    onload: function(report) {
        report.page.add_inner_button("Sales Orders", function() {
            report.data = report.raw_data.sales_orders;
            report.refresh();
        });

        report.page.add_inner_button("Sales Invoices", function() {
            report.data = report.raw_data.sales_invoices;
            report.refresh();
        });
    },
    get_data: function(data) {
        return data;
    }
};

